modules/dnn/src/model.cpp

```diff
@@ -0,0 +1,212 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+#include <algorithm>
+#include <iostream>
+#include <utility>
+#include <iterator>
+
+#include <opencv2/imgproc.hpp>
+
+namespace cv {
+namespace dnn {
+
+struct Model::Impl {
+    Size   size;
+    Scalar mean;
+    float  scale = 1.0;
+    bool   swapRB = false;
+    bool   crop = false;
+    Mat    blob;
+    std::vector<String> outNames;
```

Comment:
Maybe we can add a constructor which fills output layers names by `getUnconnectedOutLayersNames` once? Because now it's used every time.

Reply:
We can also store output blobs vector here.

Reply:
done

Reply:
For now `getUnconnectedOutLayersNames` is called at `predict` what means we call it every iteration. However we can just initialize all the output layers names in the Impl constructor.

Reply:
Maybe initialize all the output layers names in Model constructor?

Reply:
@l-bat, also good. At least we need to do it one.